core/server/src/main/java/alluxio/worker/block/BlockWorkerClientServiceHandler.java

```diff
@@ -196,23 +199,40 @@ public String call() throws AlluxioException, IOException {
   }
 
   /**
-   * Used to request space for some block file.
+   * Requests space for a block.
    *
    * @param sessionId the id of the client requesting space
    * @param blockId the id of the block to add the space to, this must be a temporary block
    * @param requestBytes the amount of bytes to add to the block
    * @return true if the worker successfully allocates space for the block on block’s location,
    *         false if there is no enough space
+   * @throws AlluxioTException if an Alluxio error occurs
+   * @throws ThriftIOException if an I/O error occurs
    */
   @Override
-  public boolean requestSpace(long sessionId, long blockId, long requestBytes) {
-    try {
-      mWorker.requestSpace(sessionId, blockId, requestBytes);
-      return true;
-    } catch (Exception e) {
-      LOG.error("Failed to request {} bytes for block: {}", requestBytes, blockId, e);
-    }
-    return false;
+  public boolean requestSpace(final long sessionId, final long blockId, final long requestBytes)
+      throws AlluxioTException, ThriftIOException {
+    return RpcUtils.call(new RpcCallable<Boolean>() {
```

Comment:
According to http://programmers.stackexchange.com/questions/149563/should-we-avoid-object-creation-in-java, java GC is optimized to deal with collecting short-lived objects like the `RpcCallable` here.

Additionally, escape analysis should make it possible for the JVM to allocate the RpcCallable on the stack, making the object allocation/deallocation essentially free. From http://www.ibm.com/developerworks/java/library/j-jtp09275/index.html,

> Escape analysis
The Java language does not offer any way to explicitly allocate an object on the stack, but this fact doesn't prevent JVMs from still using stack allocation where appropriate. JVMs can use a technique called escape analysis, by which they can tell that certain objects remain confined to a single thread for their entire lifetime, and that lifetime is bounded by the lifetime of a given stack frame. Such objects can be safely allocated on the stack instead of the heap. Even better, for small objects, the JVM can optimize away the allocation entirely and simply hoist the object's fields into registers.

Given that (1) this allocates one object per RPC and the time-cost of this allocation is probably dwarfed by the time the RPC takes, and (2) it's entirely possible that the object creation is free, I think we should optimize here for code organization and maintainability, and only revisit this if we benchmark and find that these objects are actually having an impact.